lib/serializer.ts

```diff
@@ -683,7 +684,9 @@ function getPolymorphicMapper(serializer: Serializer, mapper: CompositeMapper, o
 }
 
 function getPolymorphicDiscriminatorRecursively(serializer: Serializer, mapper: CompositeMapper): PolymorphicDiscriminator | undefined {
-  return  mapper.type.polymorphicDiscriminator || (mapper.type.uberParent && serializer.modelMappers[mapper.type.uberParent].type.polymorphicDiscriminator);
+  return  mapper.type.polymorphicDiscriminator
+    || (mapper.type.uberParent && serializer.modelMappers[mapper.type.uberParent] && serializer.modelMappers[mapper.type.uberParent].type.polymorphicDiscriminator)
+    || (mapper.type.className && serializer.modelMappers[mapper.type.className] && serializer.modelMappers[mapper.type.className].type.polymorphicDiscriminator);
```

Comment:
Refactor these expressions to a function that takes a property name. (Remove the duplication, please).